File: Webapp/Api/PC_Web_Shop/Endpoints/KorisnikEndpoints/KupacEndpoints/KupacGetAll/KupacGetAllEndpoint.cs

using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;
using PC_Web_Shop.Data;
using PC_Web_Shop.Helper;
using PC_Web_Shop.Helper.Services;

namespace PC_Web_Shop.Endpoints.KorisnikEndpoints.KupacEndpoints.KupacGetAll
{
    [Route ("kupac")]
    public class KupacGetAllEndpoint:MyBaseEndpoint<NoRequest,KupacGetAllResponse>
    {
        private readonly ApplicationDbContext _applicationDbContext;
        public KupacGetAllEndpoint(ApplicationDbContext applicationDbContext)
        {
            _applicationDbContext = applicationDbContext;
        }
        [HttpGet("kupac-get-all")]
        public override async Task<KupacGetAllResponse> Obradi([FromQuery]NoRequest noRequest,CancellationToken cancellationToken)
        {
            var kupci = await _applicationDbContext.Kupac.OrderByDescending(x => x.Id)
                .Select(x=>new KupacGetAllResponseKupci()
                {
                    Id = x.Id,
                    BrojMobitela = x.BrojMobitela,
                    Email = x.Email,
                    GradId = x.GradId,
                    Ime = x.Ime,
                    KorisnickoIme   = x.KorisnickoIme,
                    Prezime = x.Prezime,
                    SlikaKorisnika=x.SlikaKorisnika
                }).ToListAsync(cancellationToken:cancellationToken);

            return new KupacGetAllResponse { 
            Kupci = kupci,
            };
        }
    }
}

File: Webapp/Front-end/src/app/endpoints/kupac-endpoints/Kupac-getall.endpoint.ts
import {MyBaseEndpoint} from "../MyBaseEndpoint";
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {MojConfig} from "../../moj-config";
import {Injectable} from "@angular/core";
@Injectable({providedIn: 'root'})
export class KupacGetallEndpoint implements MyBaseEndpoint<void, KupacGetAllResponse>{
  constructor(public httpClient:HttpClient) {
  }
  obradi(request: void): Observable<KupacGetAllResponse> {
    let url = MojConfig.adresa_servera+'/kupac/kupac-get-all';

    return this.httpClient.get<KupacGetAllResponse>(url);
  }

}
export interface KupacGetAllResponse{
  kupci:KupacGetAllResponseKupci[];
}
export interface KupacGetAllResponseKupci {
  id: number,
  korisnickoIme: string,
  slikaKorisnika: string,
  ime: string,
  prezime: string,
  email: string,
  brojMobitela: string,
  gradId: number
}
